cuqi/experimental/mcmc/_utilities.py

```diff
@@ -1,17 +1,164 @@
+        as_string : boolean
+            Whether to return the name of the samplers in the sampling strategy as a string instead of instantiating samplers. *Optional*
+    
+
+    """
+
+    if not isinstance(target, cuqi.distribution.JointDistribution):
+        return None
+
+    par_names = target.get_parameter_names()
+
+    valid_samplers = dict()
+    for par_name in par_names:
+        conditional_params = {par_name_: np.ones(target.dim[i]) for i, par_name_ in enumerate(par_names) if par_name_ != par_name}
+        conditional = target(**conditional_params)
+
+        samplers = find_valid_samplers(conditional, as_string)
+        if len(samplers) == 0:
+            return None
+        
+        valid_samplers[par_name] = samplers
+
+    return valid_samplers
+
+def suggest_sampler(target, as_string = False, exceptions = []):
+    """
+        Suggests a possible sampler that can be used for sampling from the target distribution.
+        Return None if no sampler could be suggested.
+
+        Parameters
+        ----------
+
+        target : `cuqi.distribution.Distribution`
+            The target distribution to sample.
+        
+        as_string : boolean
+            Whether to return the name of the sampler as a string instead of instantiating a sampler. *Optional*
+    
+        exceptions : list of cuqi.experimental.mcmc sampler classes
+            Samplers not to consider for suggestion. *Optional*
+
+    """
+
+    # TODO: Conditions for suggestions, e.g., only use CWMH when the dimension of the problem is low
+    # TODO: Discuss ordering
+
+    # Samplers with suggested default values (when no defaults are defined)
+    ordering = [
+        # Direct and Conjugate samplers
+        (samplers.Direct, {}),
+        (samplers.Conjugate, {}),
+        (samplers.ConjugateApprox, {}),
+        # Specialized samplers
+        (samplers.LinearRTO, {}),
+        (samplers.RegularizedLinearRTO, {}),
+        (samplers.UGLA, {}),
+        # Gradient.based samplers (Hamiltonian and Langevin)
+        (samplers.NUTS, {}),
+        (samplers.MALA, {}),
+        (samplers.ULA, {}),
+        # Gibbs and Componentwise samplers
+        (samplers.HybridGibbs, {"sampling_strategy" : suggest_sampling_strategy(target, as_string = False)}),
+        (samplers.CWMH, {"scale" : 0.05*np.ones(target.dim),
+                         "x0" : 0.5*np.ones(target.dim)}),
+        # Proposal based samplers
+        (samplers.PCN, {"scale" : 0.02}),
+        (samplers.MH, {}),
+    ]
+
+    valid_samplers = find_valid_samplers(target, as_string = False)
+    
+    for suggestion, values in ordering:
+        if suggestion in valid_samplers and suggestion not in exceptions:
+            # Sampler found
+            if as_string:
+                return suggestion.__name__
+            else:
+                return suggestion(target, **values)
+
+    # No sampler can be suggested
+    return None
+
+def suggest_sampling_strategy(target, as_string = False):
```

Comment:
Here I also recommend the name: recommended_HybridGibbs_sampling_strategy. Using the word recommend to indicate this is what we think is preferred, and using HybridGibbs for clarity.